scripts/postInstall.js

```diff
@@ -108,72 +108,86 @@ module.exports = (context) => {
         const parseString = xml2js.parseString;
         const manifestPath = androidManifestPath + '/AndroidManifest.xml';
         const androidManifest = fs.readFileSync(manifestPath).toString();
-
+        const configFilePath = context.opts.projectRoot + '/config.xml';
         if (androidManifest) {
             parseString(androidManifest, (err, manifest) => {
 
-                if (err) return console.error(err);
+                if (err) return console.error('Error parsing AndroidManifest', err);
 
                 // Getting the package name from the manifest
-                const packageName = manifest['manifest']['$']['package']
-                const packagePath = packageName.replace(/\./g, "/");
-
-                // HyperActivity
-                const hyperActivityInputPath = utilities.getAndroidSourcePath(context) + "/in/juspay/hypersdk/HyperActivity.java";
-                const hyperActivityOutputPath = utilities.getAndroidSourcePath(context) + "/" + packagePath + '/HyperActivity.java';
-
-                // Replacing packageName in HyperActivity with app's packageName and writing it to the merchant's packagePath
-                let hyperActivityCode = fs.readFileSync(hyperActivityInputPath).toString();
-                hyperActivityCode = hyperActivityCode.replace(/\$\{mypackage\}/g, packageName);
-                fs.writeFile(hyperActivityOutputPath, hyperActivityCode, function (err) {
-                    if (err) return console.error(err);
-                });
-
-                // deleting the old HyperActivity
-                fs.unlink(hyperActivityInputPath, function (err) {
-                    if (err) return console.error(err);
+                let packageName = manifest['manifest']['$']['package']
+
+                let configXmlData = fs.readFileSync(configFilePath).toString();
+                xml2js.parseString(configXmlData, (error, configData) => {
```

Comment:
We are reading config file even if are successfully able to find package name from manifest?

You can move the rest of the functionality which uses package name to function and call it in either cases (manifest/config)